subprojects/logging/src/main/java/org/gradle/api/logging/configuration/LoggingConfiguration.java

```diff
@@ -45,6 +45,17 @@
     void setConsoleOutput(ConsoleOutput consoleOutput);
 
     /**
+     * Returns whether dry-run mode is enabled or not.
+     * Defaults to false.
+     */
+    boolean isDryRun();
+
+    /**
+     * Enables or disables dry-run (log but do not perform task actions) mode.
+     */
+    void setDryRun(boolean dryRun);
```

Comment:
Is this necessary?  It seems really weird to me that dry-run would be a logging option.